Regression: 379.v080d932e61e4 doesn't support TLS v1.3

### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  <!-- Paste your environment details below -->
  ```text
Jenkins: 2.541.1
OS: Linux - 6.8.0-60-generic
Java: 25.0.1 - Eclipse Adoptium (OpenJDK 64-Bit Server VM)
---
ansicolor:536.v13fa_b_860c267
antisamy-markup-formatter:173.v680e3a_b_69ff3
apache-httpcomponents-client-4-api:4.5.14-269.vfa_2321039a_83
apache-httpcomponents-client-5-api:5.6-183.ve5a_8a_b_e71e59
asm-api:9.9.1-189.vb_5ef2964da_91
authentication-tokens:1.144.v5ff4a_5ec5c33
bootstrap5-api:5.3.8-895.v4d0d8e47fea_d
bouncycastle-api:2.30.1.82-277.v70ca_0b_877184
branch-api:2.1280.v0d4e5b_b_460ef
build-timeout:1.39
caffeine-api:3.2.3-194.v31a_b_f7a_b_5a_81
checks-api:402.vca_263b_f200e3
cloud-stats:377.vd8a_6c953e98e
cloudbees-folder:6.1079.vc0975c2de294
command-launcher:123.v37cfdc92ef67
commons-compress-api:1.28.0-2
commons-lang3-api:3.20.0-109.ve43756e2d2b_4
commons-text-api:1.15.0-210.v7480a_da_70b_9e
configuration-as-code:2037.v8e5349845172
coverage:2.3139.v731d2e83ea_8a_
credentials:1487.va_d001ede2c31
credentials-binding:717.v951d49b_5f3a_a_
data-tables-api:2.3.5-1497.v38449eb_7d5a_1
display-url-api:2.217.va_6b_de84cc74b_
docker-commons:457.v0f62a_94f11a_3
docker-java-api:3.7.0-133.v93b_8fb_c17a_77
docker-plugin:1308.vff6e33248305
docker-slaves:1.0.7
docker-workflow:634.vedc7242b_eda_7
durable-task:651.v1f5e074fc83f
echarts-api:6.0.0-1165.vd1283a_3e37d4
eddsa-api:0.3.0.1-29.v67e9a_1c969b_b_
email-ext:1933.v45cec755423f
font-awesome-api:7.1.0-882.v1dfb_771e3278
forensics-api:3.1832.va_1179842528b_
git:5.9.0
git-client:6.5.0
github:1.45.0
github-api:1.330-492.v3941a_032db_2a_
github-branch-source:1967.vdea_d580c1a_b_a_
github-checks:634.v371dc6d978a_3
github-oauth:685.v53b_070455063
gson-api:2.13.2-173.va_a_092315913c
hashicorp-vault-plugin:379.v080d932e61e4
hidden-parameter:504.v610a_b_b_c05003
htmlpublisher:427
instance-identity:203.v15e81a_1b_7a_38
ionicons-api:94.vcc3065403257
jackson2-api:2.20.1-423.v13951f6b_6532
jakarta-activation-api:2.1.4-1
jakarta-mail-api:2.1.5-1
jakarta-xml-bind-api:4.0.6-12.vb_1833c1231d3
javax-activation-api:1.2.0-8
javax-mail-api:1.6.2-11
jaxb:2.3.9-143.v5979df3304e6
jdk-tool:83.v417146707a_3d
jjwt-api:0.11.5-120.v0268cf544b_89
job-dsl:1.93
joda-time-api:2.14.0-177.vd7e9347b_e7d5
jquery3-api:3.7.1-619.vdb_10e002501a_
json-api:20251224-185.v0cc18490c62c
json-path-api:2.10.0-202.va_9cc16c1e476
jsoup:1.22.1-76.v9cdb_2456c0e3
junit:1396.v095840ed8491
ldap:807.v7d7de30930cf
mailer:525.v2458b_d8a_1a_71
matrix-auth:3.2.9
matrix-project:870.v9db_fcfc2f45b_
metrics:4.2.37-494.v06f9a_939d33a_
mina-sshd-api-common:2.16.0-167.va_269f38cc024
mina-sshd-api-core:2.16.0-167.va_269f38cc024
okhttp-api:4.12.0-195.vc02552c04ffd
pam-auth:1.12
pipeline-build-step:584.vdb_a_2cc3a_d07a_
pipeline-github-lib:65.v203688e7727e
pipeline-graph-analysis:254.v0f63a_a_447dca_
pipeline-graph-view:777.vdeb_113b_b_94da_
pipeline-groovy-lib:787.ve2fef0efdca_6
pipeline-input-step:540.v14b_100d754dd
pipeline-milestone-step:138.v78ca_76831a_43
pipeline-model-api:2.2277.v00573e73ddf1
pipeline-model-definition:2.2277.v00573e73ddf1
pipeline-model-extensions:2.2277.v00573e73ddf1
pipeline-rest-api:2.39
pipeline-stage-step:322.vecffa_99f371c
pipeline-stage-tags-metadata:2.2277.v00573e73ddf1
pipeline-stage-view:2.39
plain-credentials:199.v9f8e1f741799
plugin-util-api:6.1192.v30fe6e2837ff
prism-api:1.30.0-703.v116fb_3b_5b_b_a_a_
resource-disposer:0.25
scm-api:728.vc30dcf7a_0df5
script-security:1399.ve6a_66547f6e1
snakeyaml-api:2.5-149.v72471e9c6371
ssh-agent:386.v36cc0c7582f0
ssh-credentials:361.vb_f6760818e8c
ssh-slaves:3.1096.v0b_cc466e4323
sshd:3.384.vc89b_5e138cf9
structs:362.va_b_695ef4fdf9
timestamper:1.30
token-macro:477.vd4f0dc3cb_cf1
trilead-api:2.284.v1974ea_324382
variant:70.va_d9f17f859e0
workflow-aggregator:608.v67378e9d3db_1
workflow-api:1398.v67030756d3fb_
workflow-basic-steps:1098.v808b_fd7f8cf4
workflow-cps:4255.vd9c37f80fd8a_
workflow-durable-task-step:1464.v2d3f5c68f84c
workflow-job:1571.vb_423c255d6d9
workflow-multibranch:821.vc3b_4ea_780798
workflow-scm-step:466.va_d69e602552b_
workflow-step-api:710.v3e456cc85233
workflow-support:1015.v785e5a_b_b_8b_22
ws-cleanup:0.49
  ```

</details>


### What Operating System are you using (both controller, and any agents involved in the problem)?

Ubuntu 24.04. The controller is using the jenkins/jenkins:2.541.1-lts-jdk25 Docker image.

### Reproduction steps

1. Set up a Vault server with `tls_min_version` set to "tls13".
2. Set up Jenkins with the Hashicorp Vault plugin, using an https path to Vault.
3. Add an SSH key to Vault and a corresponding credential in Jenkins.
4. Create an SSH Agent using that credential.
5. Capture the (encrypted) traffic from the Jenkins master to the Vault server when it tries to login.

Use of an SSH credential and an SSH agent is probably not required; the issue is in logging in to Vault. This is just the first place the errors popped up for me and it help a somewhat helpful log.

### Expected Results

Jenkins should be able to connect to Vault, using TLS v1.3.

### Actual Results

On the Jenkins agent log, I get this:
```
Warning: no key algorithms provided; JENKINS-42959 disabled
SSHLauncher{host='jenkins-vm-agent1.cbf.mkat.karoo.kat.ac.za', port=22, credentialsId='agent-ssh', jvmOptions='', javaPath='', prefixStartSlaveCmd='', suffixStartSlaveCmd='', launchTimeoutSeconds=60, maxNumRetries=10, retryWaitTime=15, sshHostKeyVerificationStrategy=hudson.plugins.sshslaves.verifiers.KnownHostsFileKeyVerificationStrategy, tcpNoDelay=true, trackCredentials=true}
[02/09/26 09:39:08] [SSH] Opening SSH connection to jenkins-vm-agent1.cbf.mkat.karoo.kat.ac.za:22.
Searching for jenkins-vm-agent1.cbf.mkat.karoo.kat.ac.za in /var/jenkins_home/.ssh/known_hosts
Searching for jenkins-vm-agent1.cbf.mkat.karoo.kat.ac.za:22 in /var/jenkins_home/.ssh/known_hosts
[02/09/26 09:39:08] [SSH] SSH host key matches key in Known Hosts file. Connection will be allowed.
ERROR: could not log in into vault
com.datapipe.jenkins.vault.exception.VaultPluginException: could not log in into vault
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.VaultAppRoleCredential.getToken(VaultAppRoleCredential.java:57)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.AbstractAuthenticatingVaultTokenCredential.getToken(AbstractAuthenticatingVaultTokenCredential.java:65)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.AbstractVaultTokenCredentialWithExpiration.authorizeWithVault(AbstractVaultTokenCredentialWithExpiration.java:118)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.VaultAccessor.init(VaultAccessor.java:71)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.common.VaultHelper.getVaultSecret(VaultHelper.java:79)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.common.VaultHelper.getVaultSecretKey(VaultHelper.java:97)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.common.AbstractVaultBaseStandardCredentials.getVaultSecretKeyValue(AbstractVaultBaseStandardCredentials.java:87)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.common.VaultSSHUserPrivateKeyImpl.getUsername(VaultSSHUserPrivateKeyImpl.java:82)
	at PluginClassLoader for ssh-credentials//com.cloudbees.jenkins.plugins.sshcredentials.SSHAuthenticator.getUsername(SSHAuthenticator.java:134)
	at PluginClassLoader for ssh-credentials//com.cloudbees.jenkins.plugins.sshcredentials.impl.TrileadSSHPublicKeyAuthenticator.getRemainingAuthMethods(TrileadSSHPublicKeyAuthenticator.java:89)
	at PluginClassLoader for ssh-credentials//com.cloudbees.jenkins.plugins.sshcredentials.impl.TrileadSSHPublicKeyAuthenticator.canAuthenticate(TrileadSSHPublicKeyAuthenticator.java:81)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:196)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:214)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1950)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:147)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:588)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:574)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:683)
	at PluginClassLoader for ssh-credentials//com.cloudbees.jenkins.plugins.sshcredentials.SSHAuthenticator.newInstance(SSHAuthenticator.java:222)
	at PluginClassLoader for ssh-credentials//com.cloudbees.jenkins.plugins.sshcredentials.SSHAuthenticator.newInstance(SSHAuthenticator.java:173)
	at PluginClassLoader for ssh-slaves//hudson.plugins.sshslaves.SSHLauncher.openConnection(SSHLauncher.java:890)
	at PluginClassLoader for ssh-slaves//hudson.plugins.sshslaves.SSHLauncher.lambda$launch$0(SSHLauncher.java:446)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: io.github.jopenlibs.vault.VaultException: io.github.jopenlibs.vault.rest.RestException: javax.net.ssl.SSLHandshakeException: (protocol_version) Received fatal alert: protocol_version
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.api.OperationsBase.retry(OperationsBase.java:43)
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.api.OperationsBase.retry(OperationsBase.java:20)
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.api.Auth.loginByAppRole(Auth.java:523)
	at PluginClassLoader for hashicorp-vault-plugin//com.datapipe.jenkins.vault.credentials.VaultAppRoleCredential.getToken(VaultAppRoleCredential.java:54)
	... 29 more
Caused by: io.github.jopenlibs.vault.rest.RestException: javax.net.ssl.SSLHandshakeException: (protocol_version) Received fatal alert: protocol_version
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.rest.Rest.postOrPutImpl(Rest.java:402)
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.rest.Rest.post(Rest.java:326)
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.api.Auth.lambda$loginByAppRole$2(Auth.java:536)
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.api.OperationsBase.retry(OperationsBase.java:32)
	... 32 more
Caused by: javax.net.ssl.SSLHandshakeException: (protocol_version) Received fatal alert: protocol_version
	at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:928)
	at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:133)
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.rest.Rest.send(Rest.java:432)
	at PluginClassLoader for hashicorp-vault-plugin//io.github.jopenlibs.vault.rest.Rest.postOrPutImpl(Rest.java:396)
	... 35 more
Caused by: javax.net.ssl.SSLHandshakeException: (protocol_version) Received fatal alert: protocol_version
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:130)
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:117)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:363)
	at java.base/sun.security.ssl.Alert$AlertConsumer.consume(Alert.java:287)
	at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:202)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
	at java.base/sun.security.ssl.SSLEngineImpl.decode(SSLEngineImpl.java:734)
	at java.base/sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:689)
	at java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:504)
	at java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:480)
	at java.base/javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:673)
	at java.net.http/jdk.internal.net.http.common.SSLFlowDelegate$Reader.unwrapBuffer(SSLFlowDelegate.java:556)
	at java.net.http/jdk.internal.net.http.common.SSLFlowDelegate$Reader.processData(SSLFlowDelegate.java:452)
	at java.net.http/jdk.internal.net.http.common.SSLFlowDelegate$Reader$ReaderDownstreamPusher.run(SSLFlowDelegate.java:283)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$LockingRestartableTask.run(SequentialScheduler.java:182)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(SequentialScheduler.java:149)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(SequentialScheduler.java:207)
	... 3 more
[02/09/26 09:39:08] Launch failed - cleaning up connection
[02/09/26 09:39:08] [SSH] Connection closed.
```

Wireshark shows that the client does not support TLS v1.3:
```
            Extension: supported_versions (len=3) TLS 1.2
                Type: supported_versions (43)
                Length: 3
                Supported Versions length: 2
                Supported Version: TLS 1.2 (0x0303)
```

### Anything else?

This used to work in 371.v884a_4dd60fb_6.

### Are you interested in contributing a fix?

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -121,7 +121,7 @@
     <dependency>
       <groupId>io.github.jopenlibs</groupId>
       <artifactId>vault-java-driver</artifactId>
-      <version>6.2.0</version>
+      <version>6.2.2</version>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>

